docs/02-use-cases/06-batch-job.md

```diff
@@ -4,10 +4,121 @@ title: Batch job
 permalink: /docs/use-cases/batch-job
 ---
 
-# Batch job
-
-A lot of batch jobs are not pure data manipulation programs. For those, the existing big data frameworks are the best fit.
+## Batch job
+A lot of batch jobs are not pure data manipulation programs. For those, the existing big data frameworks are the best fit. Cadence is a more general orchestration mechanism and doesn't provide native SQL or worker data shuffle functionality out of the box, engineers wishing to rely on these would need to implement this functionality themselves.
 But if processing a record requires external API calls that might fail and potentially take a long time, Cadence might be preferable.
 
-One of our internal Uber customers use Cadence for end of month statement generation. Each statement requires calls to multiple
-microservices and some statements can be really large. Cadence was chosen because it provides hard guarantees around durability of the financial data and seamlessly deals with long running operations, retries, and intermittent failures.
+#### Use Case:
+
+One of our internal Uber customers use Cadence for end of month statement generation. Each statement requires calls to multiple microservices and some statements can be really large. Cadence was chosen because it provides hard guarantees around durability of the financial data and seamlessly deals with long running operations, retries, and intermittent failures.
+
+## Batch jobs with heartbeating
+
+Cadence is able to coordinate, restart and track progress of large batch jobs by keeping track of their incremental progress and allowing them to resume if they're stopped for any reason. This predominantly relies on the `heartbeat` feature and activity retries. 
+
+This is used in production for customers who wish to work through large batch workloads 
+
+### Considerations before starting
+
+Heartbeating cadence workers are workers who emit their progress at an appropriate interval (usually every few seconds) indicating where they are up to. Optionally, they may use progress information (like an offset number or iterator) to resume their progress. However, this necessarily implies that:
```

Comment:
> Heartbeating cadence workers are workers

probably better phrased in terms of activities, since it's individual-activity-specific.

(even if we do add worker-based heartbeating later for efficiency, it'll still affect individual activities / heartbeat data is still per activity / etc)

Reply:
oh, brainfart, yes, good catch